backend/coins/btc/transactions/transactions.go

```diff
@@ -221,6 +223,18 @@ func (transactions *Transactions) SpendableOutputs() (map[wire.OutPoint]*Spendab
 	})
 }
 
+// GetHeaderTimestamp retrieves the header timestamp for a given transaction hash.
+func (transactions *Transactions) GetHeaderTimestamp(txHash chainhash.Hash) (*time.Time, error) {
+	transactions.synchronizer.WaitSynchronized()
+	return DBView(transactions.db, func(dbTx DBTxInterface) (*time.Time, error) {
+		txInfo, err := dbTx.TxInfo(txHash)
+		if err != nil {
+			return nil, err
+		}
+		return txInfo.HeaderTimestamp, nil
```

Comment:
HeaderTimestamp is `nil` when the utxo is unconfirmed. If nil you should take the `CreatedTimestamp`, like here:

https://github.com/BitBoxSwiss/bitbox-wallet-app/blob/master/backend/coins/btc/handlers/handlers.go#L205-L208

It would probably be good to add a method `Timestamp()` to the `TransactionData` struct and add this logic there.